neps/nep-0452.md

```diff
@@ -0,0 +1,221 @@
+- The contract will then add the `pubKey1` as a function call access key with the ability to call `claim` and `create_account_and_claim`. This means that anyone with the `privKey1` (see above), can sign a transaction on behalf of this contract (signer id set to contract id) with a function call to call one of the mentioned functions to claim the assets.
+
+### Claiming a linkdrop without a NEAR Account
+
+A user with *no* wallet that is claiming the assets associated with the public key:
+
+- generates a new keypair (`pubKey2`, `privKey2`) locally. (The keypair is not written to chain at this time)
+- chooses a new account ID such as benji.near.
+- signs a transaction on behalf of the linkdrop contract (contract is set to signer_id) using `privKey1`. The method will be `create_account_and_claim`.
+- the args of this function call will contain both `pubKey2` (which will be used to create a full access key for the new account) and the account ID itself.
+- the linkdrop contract will delete the access key associated with `pubKey1` so that it cannot be used again.
+- the linkdrop contract will create the new account and transfer the funds to it alongside any other assets.
+- the user will be able to sign transactions on behalf of the new account using `privKey2`.
+
+### Claiming a linkdrop with a NEAR Account
+
+A user with an *existing* wallet that is claiming the assets associated with the public key:
+
+- signs a transaction on behalf of the linkdrop contract (contract is set to signer_id) using `privKey1`. The method will be `claim`.
+- the args of this function call will simply contain the user's existing account ID.
+- the linkdrop contract will delete the access key associated with `pubKey1` so that it cannot be used again.
+- the linkdrop contract will transfer the funds to that account alongside any other assets.
+
+
+## Reference Implementation
+
+Below are some references for linkdrop contracts.
+
+- [Link Drop Contract](https://github.com/near/near-linkdrop)
+
+- [Keypom Contract](https://github.com/keypom/keypom)
+
+
+## Security Implications
+
+1. Linkdrop Creation
+Linkdrop creation involves creating keypairs associated with mapping to assets such as $NEAR, FTs, NFTs and access keys. There are security implications with the access keys associated with the keypairs. These access keys should be function call access keys and be limited to specific functionality. For example, keypairs can be generated with function call access keys which have permission to only call `claim` and `create_account_and_claim`. Another important security implication of linkdrop creation is to ensure that only one key is mapped to a set of assets at any given time. Linkdrop creation can happen in real-time or prior to being received. The mass creation of linkdrops involves creating multiple keypairs associated with mapping to assets and access keys. The security implications mentioned above are increased with the mass creation of linkdrops.
```

Comment:
This paragraph is not clear: is it about the fact of assigning the `pubKey1` to the contract account?